### PR TITLE
release: Add v1.0.0-beta2 release notes

### PR DESCRIPTION
**Pull request by marclop**
## Description

Adds the release notes for the upcoming release. Additionally, splits
the release notes into separate different files so that adding notes
becomes easier.

## Types of Changes

- [x] Documentation

### CODE REVIEW

File: docs/release_notes/v1.0.0-beta2.adoc [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+* *--kind flag autocompletion*`. Commands which have a `--kind` flag, now have shell autocompletion for the allowed kinds for the command.
+
+* *Improved documentation*. We now include the full {p} command reference with our official documentation. We also added auto-completion instructions to the docs.
+
+* *Added an `--all` flag for all `resync` commands*. Allowing to trigger a resync not only for a specific id but for all of the indexed records.
+
+* *Changes the default polling period for the `platform allocator vacate` operation*. Defaults to 10s instead of 1s. This change should reduce the computational impact that the `allocator vacate` operation has in both the machine running the command and the target environment where the vacate is being run.
+
+* *Adds --max-poll-retries and --poll-frequency flags to `platform allocator vacate`*. Allowing users to configure these settings if needed.
```
Comment (nrichers) [inline docs/release_notes/v1.0.0-beta2.adoc]:
```suggestion
* *Added `--max-poll-retries` and `--poll-frequency` flags to the `platform allocator vacate` command*. If you need these settings, you can now configure them.
```

(Is there anything else you can add here, even if it's just to say what the flags do in plain English?)

File: docs/release_notes/v1.0.0-beta2.adoc [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+* *Adds a `--plan-history` flag to `deployment show`*. To be able to obtain the deployment plan history from a specific deployment.
```
Comment (nrichers) [inline docs/release_notes/v1.0.0-beta2.adoc]:
```suggestion
* *Added a `--plan-history` flag to the `deployment show` command*. This flag lets you obtain the deployment plan history for a specific deployment.
```

File: docs/release_notes/v1.0.0-beta2.adoc [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+[float]
+[id="{p}-release-notes-v1.0.0-beta2-bug-fixes"]
+==== Bug fixes
+
+* `ecctl init` returned a cryptic error when the API Key could not be verified. It now returns a human-readable error which the user can act on.
```
Comment (nrichers) [inline docs/release_notes/v1.0.0-beta2.adoc]:
```suggestion
* `ecctl init` returned a cryptic error when the API Key could not be verified. It now returns a human-readable error which you can act on.
```

(We should address audiences directly, it's part of our human and friendly tone.)

File: notes/v1.0.0-beta2.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
+# Changelog
```
Comment (nrichers) [inline notes/v1.0.0-beta2.md]: Would it maybe make sense to remove this Markdown file or replace the content with a pointer to the official docs? Ideally, we'd have this information in just one place and get people to look in the official docs. 

(If not, it would be good to include the suggested edits I made for the Asciidoctor version.)

File: docs/release_notes/v1.0.0-beta2.adoc [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+* `--kind` flag was incorrectly marked as optional in the `deployment plan cancel` command.
+
+[float]
+[id="{p}-release-notes-v1.0.0-beta2-breaking-changes"]
+==== Breaking changes
+
+* *Renamed --type flag to --kind*. We renamed the `--type` flag to `--kind` on the {p} where applicable to keep the CLI flag names closer to what the API declares. Commands affected by this change are:
```
Comment (nrichers) [inline docs/release_notes/v1.0.0-beta2.adoc]:
```suggestion
* *Renamed `--type` flag to `--kind`*. We renamed the `--type` flag to `--kind` for {p} where applicable to keep the CLI flag names closer to what the API declares. Commands affected by this change are:
```

File: docs/release_notes/v1.0.0-beta2.adoc [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+* *New Commands*. We added a number of new commands to implement API functionality:
++
+--
+* Deployment appsearch commands:
+
+`ecctl deployment appsearch create`:: Creates an AppSearch instance.
+`ecctl deployment appsearch delete`:: Deletes a previously shut down AppSearch deployment resource.
+`ecctl deployment appsearch show`:: Shows the specified AppSearch deployment.
+`ecctl deployment appsearch shutdown`:: Shuts down an AppSearch deployment.
+`ecctl deployment appsearch upgrade`:: Upgrades an AppSearch deployment to the Elasticsearch deployment version.
```
[the remark this answers is not in the document]
Comment (marclop) [reply docs/release_notes/v1.0.0-beta2.adoc]: This makes sense, thanks for correcting it!

File: docs/release_notes/v1.0.0-beta2.adoc [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+`ecctl platform runner list`::  Lists the existing platform runners
+`ecctl platform runner resync`:: Resynchronizes the search index and cache for the selected runner or all
+`ecctl platform runner search`:: Performs advanced runner searching
+`ecctl platform runner show`:: Shows information about the specified runner
+--
+
+* *--kind flag autocompletion*`. Commands which have a `--kind` flag, now have shell autocompletion for the allowed kinds for the command.
+
+* *Improved documentation*. We now include the full {p} command reference with our official documentation. We also added auto-completion instructions to the docs.
```
[the remark this answers is not in the document]
Comment (marclop) [reply docs/release_notes/v1.0.0-beta2.adoc]: We should be ready to go after this PR is merged.

File: docs/release_notes/v1.0.0-beta2.adoc (hunk 1 above)
Comment (marclop) [reply docs/release_notes/v1.0.0-beta2.adoc]: So these flags basically modify the behavior of the plan poller which is used to monitor a deployment's change via the pending plan. This means that `--max-poll-retries` modifies the amount of API failures to tolerate (due to 404 because the pending plan has finished or because there's been some network errors), while `--poll-frequency` controlls how often the pending plan is obtained and then streamed to the user's terminal output, which now defaults to 10s on the vacate operations.

I hope this clarifies a bit what the flags do.

File: notes/v1.0.0-beta2.md (hunk 4 above)
Comment (marclop) [reply notes/v1.0.0-beta2.md]: It might be a good idea to redirect people to the release notes directly with an anchor to the version specific contents. I think we should do that from now on.
